sceptre_cdk_handler/cdk_builder.py

```diff
@@ -146,45 +178,175 @@ def _only_asset_is_template(self, asset_artifacts: aws_cdk.cx_api.AssetManifestA
+
+        This class is useful for deploying non-Python CDK projects with Sceptre.
+
+        Args:
+            logger: The Template Handler's logger
+            connection_manager: The Template Handler's ConnectionManager
+            cdk_json_path: The Path to the cdk.json file
+            bootstrapless_config: The configurations (in snake_case) used for the bootstrapless
+                synthesizer.
+            stack_logical_id: The LogicalID of the stack to be deployed as it is configured on the
+                App in the CDK project.
+            subprocess_run: A callable used to run subprocesses
+        """
+        super().__init__(
+            logger,
+            connection_manager,
+            subprocess_run=subprocess_run,
+        )
+        self._cdk_json_path = cdk_json_path
+        self._stack_logical_id = stack_logical_id
+        self._bootstrapless_config = bootstrapless_config
+
+    def build_template(self, cdk_context: Optional[dict], sceptre_user_data: Any):
+        if sceptre_user_data:
+            self._logger.warning(
+                "The cdk_json deployment_type does not support sceptre_user_data. Any values passed "
+                "to your stack must be done via the cdk context. All values in your sceptre_user_data "
+                "will be ignored."
+            )
+
+        environment_variables = self._get_envs()
+        if self._bootstrapless_config:
+            self._add_bootstrapless_envs(environment_variables)
+
+        with TemporaryDirectory() as output_dir:
+            self._synthesize(output_dir, cdk_context, environment_variables)
+            assets_manifest = self._get_assets_manifest(output_dir)
+            if self._only_asset_is_template(assets_manifest):
+                # Sceptre already has a mechanism to upload the template if configured. We don't
+                # need to deploy assets if the only asset is the template
+                self._logger.debug("Only asset is template; Skipping asset upload.")
+            else:
+                assets_file = Path(output_dir, f'{self._stack_logical_id}.assets.json')
+                self._publish_artifacts(str(assets_file), environment_variables)
+
+            template_file = Path(output_dir, f'{self._stack_logical_id}.template.json')
+            template = self._get_template(template_file)
+            return template
+
+    def _synthesize(self, output_dir: str, cdk_context: Optional[dict], envs: Dict[str, str]):
+        command = self._create_synth_command(output_dir, cdk_context)
+        # Run the synth with the cwd of the cdk.json's directory
+        self._run_command(command, envs, str(self._cdk_json_path.parent.resolve()))
+
+    def _create_synth_command(self, output_dir: str, cdk_context: Dict[str, str]):
+        command = f'npx cdk synth {self._stack_logical_id} -o {output_dir} -q '
+        for key, value in cdk_context.items():
+            command += f'--context {key}={value} '
```

Comment:
```suggestion
            command += f'--context {key}="{value}" '
```

The context value needs quoting to cope with spaces in values.